### PR TITLE
[WIP] Add symfony 3.0 support

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR adds symfony 3.0 support.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,28 +15,38 @@
             "homepage": "https://github.com/sonata-project/SonataCoreBundle/contributors"
         }
     ],
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/Ener-Getick/PHP-CS-Fixer"
+        },
```
Comment (Ener-Getick) [inline composer.json]: Must be removed once https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1510 is merged.

File: composer.json [hunk 2]
```diff
@@ -15,28 +15,38 @@
...
+        {
+            "type": "vcs",
+            "url": "https://github.com/Ener-Getick/exporter"
+        }
```
Comment (Ener-Getick) [inline composer.json]: Must be removed once https://github.com/sonata-project/exporter/pull/55 is merged.

File: Form/Type/DoctrineORMSerializationType.php [hunk 3]
```diff
@@ -137,10 +137,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         }
     }
 
+    // BC for SF < 3.0
```
Comment (Soullivaneuh) [inline Form/Type/DoctrineORMSerializationType.php]: `{@inheritdoc}` should be added with your comment after it.

Please do it for other classes.

File: Tests/Form/Type/BooleanTypeTest.php [hunk 4]
```diff
@@ -85,11 +85,11 @@ public function testOptions()
         ), $resolvedOptions);
     }
 
-    public function testDeprecatedCatalogueOption()
+    public function testLegacyCatalogueOption()
```
Comment (Soullivaneuh) [inline Tests/Form/Type/BooleanTypeTest.php]: Why use `configureOptions` and rename this method to legacy?

IMO, both tests should be here.

File: Tests/Form/Type/DoctrineORMSerializationTypeTest.php [hunk 5]
```diff
@@ -63,7 +64,12 @@ public function testBuildForm()
         $metadataFactory = $this->getMetadataFactoryMock();
         $registry = $this->getRegistryMock();
 
-        $type = new DoctrineORMSerializationType($metadataFactory, $registry, 'form_type_test', $this->class, 'serialization_api_write');
+        if (!method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
```
Comment (Soullivaneuh) [inline Tests/Form/Type/DoctrineORMSerializationTypeTest.php]: For clearer logic, I would check if method_exists first. If not, other statement.

File: Tests/Form/Type/DoctrineORMSerializationTypeTest.php [hunk 6]
```diff
@@ -177,4 +193,11 @@ protected function getRegistryMock($identifiers = array())
 
         return $registry;
     }
+
```
Comment (Soullivaneuh) [inline Tests/Form/Type/DoctrineORMSerializationTypeTest.php]: Missing phpdoc.

File: Tests/Form/Type/ImmutableArrayTypeTest.php [hunk 7]
```diff
@@ -22,10 +22,10 @@ public function testGetDefaultOptions()
         $type = new ImmutableArrayType();
 
         $this->assertEquals('sonata_type_immutable_array', $type->getName());
-        $this->assertEquals('form', $type->getParent());
+        $this->assertEquals(!method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ? 'form' : 'Symfony\Component\Form\Extension\Core\Type\FormType', $type->getParent());
```
Comment (Soullivaneuh) [inline Tests/Form/Type/ImmutableArrayTypeTest.php]: Same here, reverse the condition.

File: Tests/Form/Type/BooleanTypeTest.php (hunk 4 above)
Comment (Ener-Getick) [inline Tests/Form/Type/BooleanTypeTest.php]: I renamed the function with ``Legacy`` to respect the ``symfony/phpunit-bridge`` convention.
This function tests the legacy option ``catalogue``, using ``configureOptions`` is not related to the name of the function.

And what do you mean by "both tests" ?

Comment (Soullivaneuh) [inline Tests/Form/Type/BooleanTypeTest.php]: > And what do you mean by "both tests" ?

Test both `setDefaultOption` and `configureOptions`.

File: Tests/Form/Type/DoctrineORMSerializationTypeTest.php (hunk 6 above)
Comment (Ener-Getick) [inline Tests/Form/Type/DoctrineORMSerializationTypeTest.php]: Done

**Comment by rande** [comment]
@Ener-Getick Sorry about this late response, but I will close this PR as it duplicate this PR: https://github.com/sonata-project/SonataCoreBundle/pull/212 . I know you put hard work on it, again, sorry about that.

**Comment by Ener-Getick** [comment]
No problem @rande but your PR doesn't add symfony 3.0 support ... 
You should copy the changes of the composer.json and .travis.yml from this PR.

**Comment by rande** [comment]
Yep, I do it step by step ;) make sure all sonata's bundles works with 2.8 branch first with no deprecated warning.

**Comment by Ener-Getick** [comment]
Ok, good luck. Do not hesitate to ping me if you need any help ;)

**Pull request closed without merging**